src/category_theory/preadditive/Mat.lean

```diff
@@ -0,0 +1,486 @@
+  (α : embedding C ⋙ L ≅ F) :
+  L ≅ lift F :=
+nat_iso.of_components
+  (λ M, (additive_obj_iso_biproduct L M) ≪≫
+    (biproduct.map_iso (λ i, α.app (M.X i))) ≪≫
+    (biproduct.map_iso (λ i, (embedding_lift_iso F).symm.app (M.X i))) ≪≫
+    (additive_obj_iso_biproduct (lift F) M).symm)
+(λ M N f, begin
+  dsimp only [iso.trans_hom, iso.symm_hom, biproduct.map_iso_hom],
+  simp only [additive_obj_iso_biproduct_naturality_assoc],
+  simp only [biproduct.matrix_map_assoc, category.assoc],
+  simp only [additive_obj_iso_biproduct_naturality'],
+  simp only [biproduct.map_matrix_assoc, category.assoc],
+  congr,
+  ext j k ⟨⟩,
+  dsimp, simp,
+  convert α.hom.naturality (f j k),
+  erw [biproduct.matrix_π],
+  simp,
+end).
+
+-- TODO is there some uniqueness statement for the natural isomorphism in `lift_unique`?
+
+/-- Two additive functors `Mat_ C ⥤ D` are naturally isomorphic if
+their precompositions with `embedding C` are naturally isomorphic as functors `C ⥤ D`. -/
+@[ext]
+def ext {F G : Mat_ C ⥤ D} [functor.additive F] [functor.additive G]
+  (α : embedding C ⋙ F ≅ embedding C ⋙ G) : F ≅ G :=
+(lift_unique (embedding C ⋙ G) _ α) ≪≫ (lift_unique _ _ (iso.refl _)).symm
+
+/--
+Natural isomorphism needed in the construction of `equivalence_self_of_has_finite_biproducts`.
+-/
+def equivalence_self_of_has_finite_biproducts_aux [has_finite_biproducts C] :
+  embedding C ⋙ 𝟭 (Mat_ C) ≅ embedding C ⋙ lift (𝟭 C) ⋙ embedding C :=
+functor.right_unitor _ ≪≫
+  (functor.left_unitor _).symm ≪≫
+  (iso_whisker_right (embedding_lift_iso _).symm _) ≪≫
+  functor.associator _ _ _
+
+/--
+A preadditive category that already has finite biproducts is equivalent to its additive envelope.
+
+Note that we only prove this for a large category;
+otherwise there are universe issues that I haven't attempted to sort out.
+-/
+def equivalence_self_of_has_finite_biproducts
+  (C : Type (u₁+1)) [large_category C] [preadditive C] [has_finite_biproducts C] :
+  Mat_ C ≌ C :=
+equivalence.mk -- I suspect this is already an adjoint equivalence, but it seems painful to verify.
+  (lift (𝟭 C))
+  (embedding C)
+  (ext equivalence_self_of_has_finite_biproducts_aux)
+  (embedding_lift_iso (𝟭 C))
+
+@[simp] lemma equivalence_self_of_has_finite_biproducts_functor
+  {C : Type (u₁+1)} [large_category C] [preadditive C] [has_finite_biproducts C] :
+  (equivalence_self_of_has_finite_biproducts C).functor = lift (𝟭 C) :=
+rfl
+
+@[simp] lemma equivalence_self_of_has_finite_biproducts_inverse
+  {C : Type (u₁+1)} [large_category C] [preadditive C] [has_finite_biproducts C] :
+  (equivalence_self_of_has_finite_biproducts C).inverse = embedding C :=
+rfl
+
+end Mat_
+
+open_locale classical -- Alternatively, we could use `DecFinType`, which doesn't yet exist.
+
+/--
+The category of matrices over a ring `R`, with objects `Fintype`.
+-/
+@[derive [category, preadditive]]
+def Mat (R : Type*) [ring R] :=
+induced_category (Mat_ (single_obj R)) (λ n : Fintype, ⟨n, λ _, punit.star⟩)
```

Comment:
The examples below are very encouraging. But I still have a little worry. Is composition in this category defeq to matrix multiplication? Is the identity defeq to the identity matrix?

Reply:
No, and I don't think it is possible to do this, and still have `Mat` be a specialization of `Mat_` (where the objects are formal direct sums of varying objects, and there's just no one to avoid have `eq_to_hom` on the diagonal).

They can be up-to-a-theorem equal, or I could just make `Mat` disconnected from `Mat_` and have it be defeq.

Reply:
I see. I think that in LTE I find it really quite convenient that they are defeq. But it's a minor issue. So I'm happy to leave it like this. We can always refactor.